front/src/applications/operationalStudies/views/SimulationResults.tsx

```diff
@@ -73,11 +76,27 @@ const SimulationResults = ({
     useState(SPEED_SPACE_CHART_HEIGHT);
   const [mapCanvas, setMapCanvas] = useState<string>();
 
-  const [projectPathTrainResult, setProjectPathTrainResult] = useState<TrainSpaceTimeData[]>([]);
+  const [projectPathTrainResult, setProjectPathTrainResult] = useState<
+    OccupancyTrainSpaceTimeData[]
+  >([]);
 
   useEffect(() => {
     if (projectionData?.projectedTrains) {
-      setProjectPathTrainResult(projectionData?.projectedTrains || []);
+      const timetableItemDict = keyBy(timetableItemsWithDetails, 'id');
+      setProjectPathTrainResult(
+        (projectionData?.projectedTrains || []).map((train) => {
+          const timetableItem = timetableItemDict[train.id];
+          return (
+            timetableItem
+              ? {
+                  ...train,
+                  originPathItemLocation: timetableItem.path.at(0),
+                  destinationPathItemLocation: timetableItem.path.at(-1),
+                }
+              : train
+          ) as OccupancyTrainSpaceTimeData;
```

Comment:
We could just do that to avoid the cast. If `timetableItem` is `undefined`, `originPathItemLocation` and `destinationPathItemLocation` will be undefined too.
```suggestion
          return {
            ...train,
            originPathItemLocation: timetableItem?.path.at(0),
            destinationPathItemLocation: timetableItem?.path.at(-1),
          };
```